images/installer/Dockerfile

```diff
@@ -1,32 +1,14 @@
-ARG CONTAINERD_RUNWASI="download-containerd-runwasi"
+FROM golang:1.21
 
-FROM ubuntu:22.04 AS download-containerd-runwasi
-RUN DEBIAN_FRONTEND=noninteractive apt-get update \
-    && DEBIAN_FRONTEND=noninteractive apt-get install -y curl
+WORKDIR /app
 
-RUN mkdir -p /release/bin/ \
-    && curl -L https://github.com/containerd/runwasi/releases/download/containerd-shim-wasmedge%2Fv0.2.0/containerd-shim-wasmedge-linux-$(uname -m | sed s/aarch64/arm64/g | sed s/x86_64/amd64/g).tar.gz | tar -xzf - -C /release/bin/ \
-    && curl -L https://github.com/containerd/runwasi/releases/download/containerd-shim-wasmtime%2Fv0.2.0/containerd-shim-wasmtime-linux-$(uname -m | sed s/aarch64/arm64/g | sed s/x86_64/amd64/g).tar.gz | tar -xzf - -C /release/bin/
+COPY go.mod go.sum ./
+RUN go mod download
 
-FROM ${CONTAINERD_RUNWASI} AS containerd_runwasi
+COPY pkg ./pkg
+COPY cmd ./cmd
+COPY *.go ./
 
-FROM ubuntu:22.04 AS deislabs_containerd-wasm-shims
+RUN go build -o kwasm-node-installer
 
-RUN DEBIAN_FRONTEND=noninteractive apt-get update \
-    && DEBIAN_FRONTEND=noninteractive apt-get install -y curl
-
-RUN mkdir /assets \
-    && curl -L https://github.com/deislabs/containerd-wasm-shims/releases/download/v0.9.1/containerd-wasm-shims-v1-lunatic-linux-$(uname -m).tar.gz  | tar -xzf - -C /assets \
-    && curl -L https://github.com/deislabs/containerd-wasm-shims/releases/download/v0.9.1/containerd-wasm-shims-v1-slight-linux-$(uname -m).tar.gz  | tar -xzf - -C /assets \
-    && curl -L https://github.com/deislabs/containerd-wasm-shims/releases/download/v0.9.1/containerd-wasm-shims-v1-spin-linux-$(uname -m).tar.gz  | tar -xzf - -C /assets \
-    && curl -L https://github.com/deislabs/containerd-wasm-shims/releases/download/v0.9.1/containerd-wasm-shims-v1-wws-linux-$(uname -m).tar.gz  | tar -xzf - -C /assets
-
-
-FROM busybox
-
-COPY script/installer.sh /script/installer.sh
-COPY --link --from=deislabs_containerd-wasm-shims /assets /assets
-COPY --link --from=containerd_runwasi /containerd-shim-wasmedge-v1 /assets/
-COPY --link --from=containerd_runwasi /containerd-shim-wasmer-v1 /assets/
-COPY --link --from=containerd_runwasi /containerd-shim-wasmtime-v1 /assets/
-CMD sh /script/installer.sh wasmedge
+CMD ["/app/kwasm-node-installer", "install"]
```

Comment:
Can we make this a multi-stage build?

```
FROM scratch
COPY --from=builder /app/kwasm-node-installer /kwasm-node-installer
```

above this line and add `as builder` to the `FROM golang:1.21` line

Reply:
Definitely! Updated the Dockerfile accordingly.

Reply:
oh also, I would set `ENTRYPOINT` as `["/kwasm-node-installer"]` so that we can override the actual command to run with with `CMD` (or `args` in Kubernetes)